pkg/cloudprovider/providers/azure/azure_cache.go

```diff
@@ -60,22 +82,53 @@ func (t *timedcache) GetOrCreate(key string, createFunc func() interface{}) (int
 		return nil, err
 	}
 	if exists {
-		return (entry.(*timedcacheEntry)).data, nil
+		return entry.(*cacheEntry), nil
+	}
+
+	// Still not found, add new entry with nil data.
+	// Note the data will be filled later by getter.
+	newEntry := &cacheEntry{
+		key:  key,
+		data: nil,
 	}
+	t.store.Add(newEntry)
+	return newEntry, nil
+}
 
-	if createFunc == nil {
-		return nil, nil
+// Get returns the requested item by key.
+func (t *timedCache) Get(key string) (interface{}, error) {
+	entry, err := t.getInternal(key)
+	if err != nil {
+		return nil, err
 	}
-	created := createFunc()
-	t.store.Add(&timedcacheEntry{
+
+	// Data is still not cached yet, cache it by getter.
+	if entry.data == nil {
```

Comment:
What would be the situation where we need key with nil value?

Reply:
This is something like lazy initialization, so we could update different keys in the cache at the same time while preventing update same key simultaneously.

entry.data is set nil here: https://github.com/kubernetes/kubernetes/pull/59520/files#diff-413059dc3744227dc0d21506f9883ad8R92